packages/expo-updates/ios/EXUpdates/Database/UpdatesDatabaseInitialization.swift

```diff
@@ -195,7 +195,7 @@ internal final class UpdatesDatabaseInitialization {
     }
 
     var db: OpaquePointer?
-    if sqlite3_open(String(latestURL.absoluteString.utf8), &db) != SQLITE_OK {
+    if sqlite3_open(latestURL.path, &db) != SQLITE_OK {
```

Comment:
To test this change, we'll probably need to do a bit more than the integration/unit tests since our test coverage is pretty low.
- Reading https://developer.apple.com/documentation/foundation/nsurl and https://stackoverflow.com/questions/16176911/nsurl-path-vs-absolutestring it seems like these are equivalent for file URLs, but my confidence is low.
- If our collective confidence is low that they're always equivalent, then one way to test would be to build an app without this change and then make the change and re-build to ensure the database file is the same (check that it is already populated when launching after the change).

There are also other calls that still use `String(latestURL.absoluteString.utf8)`, so I think we'll need to update those to prevent breakage.

Reply:
Cool, I'll test it out some more and make those calls consistent.

Regarding URLs. There is a difference between `path` and `absoluteString`.

The folder used by the simulator is 
`/Library/Developer/CoreSimulator/Devices/1D16D72D-581F-4F83-9F68-E6AAFEE047A7/data/Library/Application Support`

When using `absoluteString` you'll get this
`/Library/Developer/CoreSimulator/Devices/1D16D72D-581F-4F83-9F68-E6AAFEE047A7/data/Library/Application%20Support/some.db` and `SQLite` can't find it because of the space encoding.

With `path` you'll get the correct string, without the encoding
`/Library/Developer/CoreSimulator/Devices/1D16D72D-581F-4F83-9F68-E6AAFEE047A7/data/Library/Application Support/some.db`

Reply:
`String(url.absoluteString.utf8)` actually does nothing to the original string.
```swift
let testStr = "/some/app/directory/Applications%20Support" // prints /some/app/directory/Applications%20Support
let newStr =  String(testStr.absoluteString.utf8) // prints /some/app/directory/Applications%20Support
```
All strings in Swift 5 are `utf8` encoded and fully compatible with C. I would think stuff like this `String(testStr.absoluteString.utf8)` is here because this module was rewritten from Obj-c which uses `utf16` encoding

Reply:
i tried to search the `absoluteString` in expo-updates code, do you think whether these two places would cause regression?
- https://github.com/expo/expo/blob/e7d314738ddab2021957452a2d41818de874555f/packages/expo-updates/ios/EXUpdates/Database/UpdatesDatabaseInitialization.swift#L137
- https://github.com/expo/expo/blob/e7d314738ddab2021957452a2d41818de874555f/packages/expo-updates/ios/EXUpdates/Database/UpdatesDatabaseInitialization.swift#L198

Reply:
Hey @Kudo I've changed both those places, do you mean is it necessary to change them?

Reply:
oh sorry, i had being thinking that that change is in the test files 🙈 thanks Alan!